### ISSUE TITLE
External CMake build doesn't link properly

### ISSUE THREAD

**Issue by abarret** [IBAMR/IBAMR#1257]
When building a external project with CMake, the project compiles but does not link correctly. In particular, running `ldd main2d` gives the following:
```
	libBUNDLED_muParser.so => not found
	libmesh_dbg.so.0 => not found
	libHYPRE-2.18.2.so => not found
	libBUNDLED_muParser.so => not found
	libmesh_dbg.so.0 => not found
```

**Comment by drwells** [comment]
The way we set up dependency exports is pretty fragile and needs to be rewritten.

**Comment by drwells** [comment]
If you set up an external project and you are using the Makefile generator there should be a file named `link.txt` somewhere inside `CMakeFiles`. Could you attach it here?

**Comment by abarret** [comment]
[link.txt](https://github.com/IBAMR/IBAMR/files/6534084/link.txt)

**Comment by drwells** [comment]
Weird - it definitely includes `/home/abarret/sfw/libmesh/1.5.1/linux-dbg/lib/libmesh_dbg.so` in the link line.

**Comment by drwells** [comment]
@abarret I want to get this fixed soon and I think I know what to do - I just need to reproduce your environment. Was this with a relatively recent release of Ubuntu?

**Comment by abarret** [comment]
It failed on a release of Linux Mint 19, although I'll have to double check that it still fails later tonight.

Building and linking (and execution) works on my work desktop which is running `CentOS Linux release 7.9.2009`. Although I need to specify the location of the c++ compiler to CMake, or else I get a bunch of linker errors.

**Comment by abarret** [comment]
I can reproduce the issue on Linux Mint 19.3 on two different computers.

**Comment by drwells** [comment]
Fixed by #1273.
